GNUmakefile

```diff
@@ -67,33 +79,33 @@ $(BINDIR)/regexJIT.o : $(REGEXDIR)/regexJIT.c $(BINDIR)/.keep $(SLJIT_HEADERS) $
 	$(CC) $(CPPFLAGS) $(REGEX_CFLAGS) -c -o $@ $(REGEXDIR)/regexJIT.c
 
 $(BINDIR)/sljit_test: $(BINDIR)/.keep $(BINDIR)/sljitMain.o $(TESTDIR)/sljitTest.c $(SRCDIR)/sljitLir.c $(SLJIT_LIR_FILES) $(SLJIT_HEADERS) $(TESTDIR)/sljitConfigPre.h $(TESTDIR)/sljitConfigPost.h
-	$(CC) $(CPPFLAGS) -DSLJIT_HAVE_CONFIG_PRE=1 -I$(TESTDIR) $(CFLAGS) $(LDFLAGS) $(BINDIR)/sljitMain.o $(TESTDIR)/sljitTest.c $(SRCDIR)/sljitLir.c -o $@ -lm -lpthread
+	$(CC) -pthread $(CPPFLAGS) -DSLJIT_HAVE_CONFIG_PRE=1 -I$(TESTDIR) $(CFLAGS) $(LDFLAGS) $(BINDIR)/sljitMain.o $(TESTDIR)/sljitTest.c $(SRCDIR)/sljitLir.c -o $@ -lm $(EXTRA_LIBS)
```

Comment:
What is the difference?

Reply:
When using `-pthread` then the coprocessor and compiler are also aware that the object needs to be "thread safe" (ex:  `-D_REENTRANT` gets defined).

also it doesn't fail to link if there is no -lpthread (ex: OpenWRT); most libc have the pthread functions included in the main library and optionally use a "stub pthread" library.

Reply:
Normally libraries are defined at the end, since it is used for checking unresolved references. If pthread is a real library, this might not work.

Reply:
It will always work, unless your gcc doesn't know how to handle the `-pthread` flag.  Some really old versions of clang (google seems to think that older than 8) didn't knew how to handle that flag, but hopefully nobody using this still cares about those.

When `-pthread` is provided, and gcc is driving the linking then -lpthread is added as needed.

`-pthread` has been part of gcc since 2.7.0 (literally from last century) and glibc only created the "stub" library with release 2.34 (end of 2021) so that is how it worked until then.

Reply:
I worry more about other compilers like Intel C compiler. Defining both early and late could work?

Reply:
Btw as far as I know all command line arguments are processed by the compiler before doing anything, so pthread at the end should have its side effect.

Reply:
FWIW for a hold of clang 3.4.2 and it recognized the `-pthread` option and linked correctly as if `-lpthread` was provided, so maybe Google was referring to an old version of Apple clang which is even less likely to be relevant.

For the sake of expediency, I have removed this change though but still think is worth considering, after all even the non opensource compilers are based on clang nowadays.